photon-core/src/main/java/org/photonvision/vision/pipe/impl/Calibrate3dPipe.java

```diff
@@ -113,9 +124,128 @@ protected CameraCalibrationCoefficients process(
             return null;
         }
 
+        for (int i = 0; i < rvecs.size(); i++) {
+            // logger.info("OpenCV found board at R="+rvecs.get(i).dump()+"t="+tvecs.get(i).dump());
+        }
+
         JsonMatOfDouble cameraMatrixMat = JsonMatOfDouble.fromMat(cameraMatrix);
         JsonMatOfDouble distortionCoefficientsMat = JsonMatOfDouble.fromMat(distortionCoefficients);
 
+        var observations =
+                createObservations(in, cameraMatrix, distortionCoefficients, rvecs, tvecs, null);
+
+        // Standard deviation of results
+        try {
+            // Print calibration successful
+            logger.info(
+                    "CALIBRATION SUCCESS for res "
+                            + in.get(0).size
+                            + " (with accuracy "
+                            + calibrationAccuracy
+                            + ")! camMatrix: \n"
+                            + new ObjectMapper().writeValueAsString(cameraMatrixMat)
+                            + "\ndistortionCoeffs:\n"
+                            + new ObjectMapper().writeValueAsString(distortionCoefficientsMat)
+                            + "\n");
+        } catch (JsonProcessingException e) {
+            logger.error("Failed to parse calibration data to json!", e);
+        }
+
+        cameraMatrix.release();
+        distortionCoefficients.release();
+        rvecs.stream().forEach(Mat::release);
+        tvecs.stream().forEach(Mat::release);
+
+        return new CameraCalibrationCoefficients(
+                in.get(0).size, cameraMatrixMat, distortionCoefficientsMat, new double[0], observations);
+    }
+
+    protected CameraCalibrationCoefficients process_mrcal(
+            List<FindBoardCornersPipe.FindBoardCornersPipeResult> in) {
+        List<MatOfPoint2f> corner_locations =
+                in.stream().map(it -> it.imagePoints).map(MatOfPoint2f::new).collect(Collectors.toList());
+
+        var start = System.nanoTime();
+        MrCalResult result =
+                MrCalJNI.calibrateCamera(
+                        corner_locations, params.boardWidth, params.boardHeight, 0.0254, 640, 480, 1200);
```

Comment:
I might be mistaken here but shouldn't image height and width be variable? OpenCV uses `new Size(in.get(0).size.width, in.get(0).size.height)`

Reply:
Lmaoooooooooooooooooooo